deploy/examples/cluster.yaml

```diff
@@ -281,6 +281,31 @@ spec:
     # No values or 0 means that the operator will wait until the placement groups are healthy before unblocking the next drain.
     pgHealthCheckTimeout: 0
 
+  # csiDriverOptions defines defines CSI Driver settings applied per cluster.
+  csiDriverOptions:
+    # common settings for all CSI drivers.
+    common:
+      readAffinity:
+        # set enabled to true to enable read affinity.
+        enabled: false
+        # CRUSH location labels define which node labels to use
+        # as CRUSH location. This should correspond to the values set in
+        # the CRUSH map.
+        # Defaults to all the labels mentioned in
+        # https://rook.io/docs/rook/latest/CRDs/Cluster/ceph-cluster-crd/#osd-topology
+        # crushLocationLabels:
+        # - "kubernetes.io/hostname"
+        # - "topology.kubernetes.io/zone"
+        # - "topology.rook.io/rack"
+
+    # cephfs driver specific settings.
+    cephfs:
+      # Set CephFS Kernel mount options to use https://docs.ceph.com/en/latest/man/8/mount.ceph/#options
+      # Set to "ms_mode=secure" when connections.encrypted is enabled in CephCluster CR
```

Comment:
We should set `ms_mode=secure` automatically if encryption is enabled. They should only set this if they need to override the defaults, right?

Reply:
> We should set `ms_mode=secure` automatically if encryption is enabled. They should only set this if they need to 
override the defaults, right?

That line is from https://github.com/rook/rook/blob/f86fca5e7ad620d54798f12a10adf88baa0663fb/deploy/examples/operator.yaml#L487
That is still something user needs to do. Doing so automatically may cause existing clients to get blocked.
https://github.com/rook/rook/pull/11625. 

cc @Madhu-1

Reply:
We couldn't do it automatically in the past because of the issue with ceph clusters that might have different settings. Now with the cluster settings being independent, seems like we can set it automatically when encryption is enabled. This setting would just allow them to override the defaults.